Source/Particles/WarpXParticleContainer.cpp

```diff
@@ -699,25 +699,129 @@ WarpXParticleContainer::DepositCharge (WarpXParIter& pti, RealVector& wp,
     // Indices of the lower bound
     const Dim3 lo = lbound(tilebox);
 
+    // HACK - sort particles by bin here.
+    WARPX_PROFILE_VAR_START(blp_sort);
+    amrex::DenseBins<ParticleType> bins;
+    {
+        const Geometry& geom = Geom(lev);
+        const auto dxi = geom.InvCellSizeArray();
+        const auto plo = geom.ProbLoArray();
+        const auto domain = geom.Domain();
+
+        auto& ptile = ParticlesAt(lev, pti);
+        auto& aos   = ptile.GetArrayOfStructs();
+        auto pstruct_ptr = aos().dataPtr();
+
+        ParticleTileType ptile_tmp;
+        ptile_tmp.define(ptile.NumRuntimeRealComps(), ptile.NumRuntimeIntComps());
+        ptile_tmp.resize(ptile.numParticles());
+
+        Box box = pti.validbox();
+        box.grow(1);
+        amrex::IntVect bin_size = WarpX::sort_bin_size;
+        int ntiles = numTilesInBox(box, true, bin_size);
+
+        bins.build(ptile.numParticles(), pstruct_ptr, ntiles,
```

Comment:
Looks like this call is failing for RZ on OpenMP

Reply:
cartesian2d with OpenMP also, both here:

```
 3: ./main2d.gnu.TEST.TPROF.MTMPI.OMP.QED.OPMD.PSATD.RZ.GPUCLOCK.ex(+0x32366f) [0x5629fbc7166f]
    amrex::Assert(char const*, char const*, int, char const*) at /tmp/ci-mSgAtyuVcG/amrex//Src/Base/AMReX.H:175
 (inlined by) operator() at /tmp/ci-mSgAtyuVcG/warpx/./Source/Particles/WarpXParticleContainer.cpp:729
 (inlined by) void amrex::DenseBins<amrex::Particle<0, 0> >::build<int, WarpXParticleContainer::DepositCharge(WarpXParIter&, amrex::PODVector<double, std::allocator<double> >&, int const*, amrex::MultiFab*, int, long, long, int, int, int)::{lambda(amrex::Particle<0, 0> const&)#1}>(amrex::BinPolicy::OpenMPBinPolicy, int, amrex::Particle<0, 0> const*, int, WarpXParticleContainer::DepositCharge(WarpXParIter&, amrex::PODVector<double, std::allocator<double> >&, int const*, amrex::MultiFab*, int, long, long, int, int, int)::{lambda(amrex::Particle<0, 0> const&)#1}&&) [clone ._omp_fn.0] at /tmp/ci-mSgAtyuVcG/amrex//Src/Particle/AMReX_DenseBins.H:326
```

Reply:
https://github.com/AMReX-Codes/amrex/blob/development/Src/Particle/AMReX_DenseBins.H#L326

Reply:
I've managed to build and run the case that actually fails in CI, and it reliably exhibits the bug. I get a backtrace, but investigation is a pain because the testing scripts build in release mode, and I don't know how to manually rebuild after making changes, rather than waste time rebuilding the whole thing from scratch